pyvista/core/dataset.py

```diff
@@ -2162,35 +2162,98 @@ def cast_to_unstructured_grid(self) -> 'pyvista.UnstructuredGrid':
         alg.Update()
         return _get_output(alg)
 
-    def cast_to_pointset(self, deep: bool = False) -> 'pyvista.PointSet':
-        """Get a new representation of this object as a :class:`pyvista.PointSet`.
+    def cast_to_pointset(self, pass_cell_data: bool = False) -> 'pyvista.PointSet':
+        """Extract the nodes of this dataset as a :class:`pyvista.PointSet`.
 
         Parameters
         ----------
-        deep : bool, optional
-            When ``True`` makes a full copy of the object.  When ``False``,
-            performs a shallow copy where the points and data arrays are
-            references to the original object.
+        pass_cell_data : bool, optional
+            Run the ``cell_data_to_point_data`` filter and pass cell data
+            fields to the new pointset.
 
         Returns
         -------
         pyvista.PointSet
             Dataset cast into a :class:`pyvista.PointSet`.
 
+        Notes
+        -----
+        This will produce a deep copy of the points and point/cell data of
+        the original mesh.
+
         Examples
         --------
         >>> import pyvista
-        >>> mesh = pyvista.Sphere()
+        >>> mesh = pyvista.Wavelet()
         >>> pointset = mesh.cast_to_pointset()
         >>> type(pointset)
         <class 'pyvista.core.pointset.PointSet'>
 
         """
         pset = pyvista.PointSet()
-        pset.SetPoints(self.GetPoints())
-        pset.GetPointData().ShallowCopy(self.GetPointData())
-        if deep:
-            return pset.copy(deep=True)
+        pset.points = self.points.copy()
+        if pass_cell_data:
+            self = self.cell_data_to_point_data()
+        pset.GetPointData().DeepCopy(self.GetPointData())
+        pset.active_scalars_name = self.active_scalars_name
+        return pset
+
+    def cast_to_poly_points(self, pass_cell_data: bool = False) -> 'pyvista.PolyData':
+        """Extract the nodes of this dataset as a :class:`pyvista.PolyData`.
```

Comment:
Would prefer moving away from "nodes" to points. I think this comes from FEA and we should generalize PyVista.

```suggestion
        """Extract the points of this dataset and return a :class:`pyvista.PolyData`.
```